src/sentry/issues/status_change.py

```diff
@@ -0,0 +1,115 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any, Dict, Sequence, Tuple
+
+from sentry.models import (
+    Activity,
+    Group,
+    GroupStatus,
+    GroupSubscription,
+    Project,
+    User,
+    record_group_history_from_activity_type,
+)
+from sentry.notifications.types import GroupSubscriptionReason
+from sentry.signals import issue_ignored, issue_unignored, issue_unresolved
+from sentry.tasks.integrations import kick_off_status_syncs
+from sentry.types.activity import ActivityType
+
+
+def handle_status_update(
```

Comment:
Just curious, are you planning to extra out the inner logic for each status as well?